cupy/core/_dtype.pyx

```diff
@@ -0,0 +1,27 @@
+import numpy
+import six
+
+
+cdef dict _dtype_dict
+
+
+cdef _init_dtype_dict():
+    global _dtype_dict
+    _dtype_dict = {i: numpy.dtype(i)
+                   for i in six.integer_types + (float, bool, complex, None)}
+    for i in 'dfDFeqlihbQLIHB?':
```

Comment:
Could you explain reason of this order?  Is it good idea to add strings of all supported types, like `typecodes` in https://github.com/numpy/numpy/blob/master/numpy/core/numerictypes.py?